go/internal/feast/onlinestore/egvalkeyonlinestore.go

```diff
@@ -354,9 +355,297 @@ func (v *ValkeyOnlineStore) OnlineRead(ctx context.Context, entityKeys []*types.
 	return results, nil
 }
 
-func (v *ValkeyOnlineStore) OnlineReadRange(ctx context.Context, groupedRefs *model.GroupedRangeFeatureRefs) ([][]RangeFeatureData, error) {
-	// TODO: Implement OnlineReadRange
-	return nil, errors.New("OnlineReadRange is not supported by ValkeyOnlineStore")
+func valkeyBatchHMGET(
+	ctx context.Context,
+	client valkey.Client,
+	entityKeyBin []byte,
+	members [][]byte,
+	fields []string,
+	fv string,
+	grp *fvGroup,
+	results [][]RangeFeatureData,
+	eIdx int,
+) error {
+	for start := 0; start < len(members); start += PIPELINE_BATCH_SIZE {
+		end := start + PIPELINE_BATCH_SIZE
+		if end > len(members) {
+			end = len(members)
+		}
```

Comment:
Condense to one line

```suggestion
		end := max(start + PIPELINE_BATCH_SIZE, len(members))
```